src/Loader/ArrayLoader.php

```diff
@@ -28,11 +28,12 @@
  * @phpstan-type TBackgroundHash array{title?: string|null, keyword?: string, line?: int, steps?: array<int, TStepHash>}
  * @phpstan-type TScenarioHash array{type?: 'scenario', title?: string|null, tags?: list<string>, keyword?: string, line?: int, steps?: array<int, TStepHash>}
  * @phpstan-type TOutlineHash array{type: 'outline', title?: string|null, tags?: list<string>, keyword?: string, line?: int, steps?: array<int, TStepHash>, examples?: TExampleTableHash|array<array-key, TExampleHash>}
- * @phpstan-type TExampleHash array{table: TExampleTableHash, tags?: list<string>}|TExampleTableHash
+ * @phpstan-type TExampleHash TTaggedExampleTableHash|TExampleTableHash
  * @phpstan-type TExampleTableHash array<int<1, max>, list<string>>
+ * @phpstan-type TTaggedExampleTableHash array{table: TExampleTableHash, tags?: list<string>}
```

Comment:
Can we find a better name for `TTaggedExampleTableHash`? I named it like that because it's a table "that could have tags" (and I needed a type to do some cleaning and referring to it).

But when you look at, for example, `isTaggedExampleTableHash()`, the name of the method kinda doesn't match with what the body is doing (namely just checking for 'table'). It would be best to keep the body as is, to stay backword compatible, but maybe consider finding a better name.

Reply:
OK, then since this works for the moment, let's keep it as proposed in this PR and when PHPStan gets updated and starts complaining, we simply fix it. This would also applies to `mb_convert_encoding()`.

What do you think?

Reply:
@stof / @acoulton ok, so is it fine to rollback the helper methods I added and just tell PHPStan to ignore the errors? Or should I improve them as per [the other comment](https://github.com/Behat/Gherkin/pull/368/files#r2141124604)?

Reply:
I think if it only affects ArrayLoader then my vote is to just tell PHPStan to ignore the errors if that's easiest - thanks.

Reply:
Done, please reopen if you have any further feedback.